### PR TITLE
Add new `status` key to events

### LINKED ISSUE #368: Canceled or postponed events
Hi! I see lots of events canceled because of the COVID-19. Wouldn't be a good idea to remove or strike through or put a comment in all these canceled events?

### PR DESCRIPTION
**Pull request by pjmartorell**
Add `status` key in order to inform about canceled or postponed events.
Resolves #368

### CODE REVIEW

**Comment by zverok** [comment]
Since the inception of the PR, also canceled:
* RubyWine
* RubyNess
* RubyConf India

**Comment by zverok** [comment]
* [x] RubyConfBY postponed to Jul 18
* [x] BrightonRuby cancelled

@jonallured We really need this notice :)

File: js/jquery.js [hunk 1]
```diff
@@ -3,4 +3,4 @@
...
...sion|file|res|widget):$/,Zt=/^(?:GET|HEAD)$/,en=/^\/\//,tn=/^([\w.+-]+:)(?:\/\/([^\/?#:]*)(?::(\d+)|)|)/,nn=x.fn.load,rn={},on={},sn="*/".concat("*");try{Ut=i.href}catch(an){Ut=o.createElement("a"),Ut.href="",Ut=Ut.href}Xt=tn.exec(Ut.toLowerCase())||[];function un(e){return function(t,n){"string"!=typeof t&&(n=t,t="*");var r,i=0,o=t.toLowerCase().match(w)||[];if(x.isFunction(n))while(r=o[i++])"+"===r[0]?(r=r.slice(1)||"*",(e[r]=e[r]||[]).unshift(n)):(e[r]=e[r]||[]).push(n)}}function ln(e,t,n,r){var i={},o=e===on;function s(a){var u;return i[a]=!0,x.each(e[a]||[],function(e,a){var l=a(t,n,r);return"string"!=typeof l||o||i[l]?o?!(u=l):undefined:(t.dataTypes.unshift(l),s(l),!1)}),u}return s(t.dataTypes[0])||!i["*"]&&s("*")}function cn(e,t){var n,r,i=x.ajaxSettings.flatOptions||{};for(n in t)t[n]!==undefined&&((i[n]?e:r||(r={}))[n]=t[n]);return r&&x.extend(!0,e,r),e}x.fn.load=function(e,t,n){if("string"!=typeof e&&nn)return nn.apply(this,arguments);var r,i,o,s=this,a=e.indexOf(" ");return a>=0&&(r=e.slice(a),e=e.slice(0,a)),x.isFunction(t)?(n=t,t=undefined):t&&"object"==typeof t&&(i="POST"),s.length>0&&x.ajax({url:e,type:i,dataType:"html",data:t}).done(function(e){o=arguments,s.html(r?
```
Comment (filipewl) [inline js/jquery.js]: What exactly has changed in here? It's hard to tell because the file is minified.

Comment (pjmartorell) [reply js/jquery.js]: @filipewl sorry, I think I modified jquery.js in error when doing a find and replace. I fix it 👍

**Comment by pjmartorell** [comment]
@filipewl Now it's fixed

**Comment by pjmartorell** [comment]
@filipewl maybe we could add a date in case it's postponed and there is another official date announced, but maybe we can add it later

**Comment by jonallured** [comment]
@pjmartorell thanks for taking the time to work through this!!

Yeah, it has been a crazy few weeks and I have nothing but sympathy for conference organizers making the tough choices about what to do with their events. I'm sure it's not easy!

Having said that I guess I'm not totally convinced these changes are worth it. For an event that is canceled, I would assume that event would simply not appear on the list. What's the value in having an entry for an event that is not going to occur?

For an event that has been postponed with a new date, I would expect to see that event on the list with that new date. I'm not sure what the value would be in marking it as postponed in that case either. 🤔 

@pjmartorell can you help me understand how you were thinking about this? Do you see value that I'm not understanding?

**Comment by zverok** [comment]
> Having said that I guess I'm not totally convinced these changes are worth it. For an event that is canceled, I would assume that event would simply not appear on the list. What's the value in having an entry for an event that is not going to occur?

I have those reasons in my head:

1. Despite the best effort, rubyconferences.org not always includes all possible conferences. For example, 2020's [WrocLove.rb](https://wrocloverb.com/) is not yet on the list. Considering the fact that for many Rubyists your site is the first go-to site for the conferences, it is hard to distinguish "It was never mentioned this year" from "it was there, just canceled".
2. Kinda idealistic one, but I believe in those uncertain times it is a valuable community service to still provide a list of "what was planned yet canceled" instead of just removing everything (which looks kinda... scary). At least so the potential visitors still could reach the conference's site/Twitter.

**Comment by adarsh** [comment]
Bumping this to revive the discussion.

My take:

- I could see both sides of this: this was initiated at the beginning of the pandemic, when this information would have been most useful. 
- Having lots of canceled events on the page seems not-great, although I believe the volume of this is now extremely low
- I think having an optional `status` attribute seems like a good change with potential use in the future.

I'm a 👍🏾 on this, after rebasing. Other thoughts?

**Comment by marcoroth** [comment]
Agreed. I also think it would make sense to add the `status` field so we can add the events that were cancelled, and for any future events the might get cancelled.

**Comment by marcoroth** [comment]
Alright, I rebased this commit and filled back some of the conferences which were removed because they were canceled. I used this pull request https://github.com/ruby-conferences/ruby-conferences.github.io/pull/379, the comments here https://github.com/ruby-conferences/ruby-conferences.github.io/pull/369#issuecomment-598349676 and the conference websites to fill in some of the missing data.

I also changed the location to "Online" if the conference was held online and wasn't canceled all-together.

The conferences are now being shown with their status next to them:

CleanShot 2023-09-12 at 23 38 34

File: README.md [hunk 2]
```diff
@@ -47,6 +47,7 @@ Extra keys for the upcoming events:
 * `reg_date`: If there is a registration deadline, enter that here - ISO8601 formatted (yyyy-mm-dd).
 * `cfp_phrase`: Typically you want to put "CFP open" here. If you also provide a `cfp_date` then you may prefer to write "CFP closes" so the site will render for example "CFP closes in 17 days".
 * `cfp_date`: If there is a cfp deadline, enter that here - ISO8601 formatted (yyyy-mm-dd).
+* `status`:  Typically you want to put "Canceled" or "Postponed" here.
```
Comment (adarsh) [inline README.md:50]: Maybe add the "online" example here also?

**Comment by adarsh** [comment]
Me personally, I would do the `status` change separately from the data change, or at least in separate commits, but it's not enough to hold up merging, IMO.

**Pull request merged**